src/elliptic/curves/ed25519.rs

```diff
@@ -326,7 +326,8 @@ impl ECPoint for Ed25519Point {
     }
 
     fn from_coords(x: &BigInt, y: &BigInt) -> Result<Ed25519Point, NotOnCurve> {
-        let expected_x = xrecover(y);
+        let is_odd = x.is_odd();
+        let expected_x = xrecover(y, x.is_odd());
```

Comment:
You can reuse variable you just introduced:

```suggestion
        let expected_x = xrecover(y, is_odd);
```

Reply:
Thanks!